compiler/modules/front/codeGen.bal

```diff
@@ -1113,12 +1113,10 @@ function codeGenTypeCast(CodeGenContext cx, bir:BasicBlock bb, Environment env,
     var reg = <bir:Register>operand; // const folding should have got rid of the const
     t:SemType toType = tcExpr.semType;
     t:UniformTypeBitSet? toNumType = t:singleNumericType(toType);
-    if toNumType != ()
-       // Remove the next line to enable conversion when support in backend is implemented
-       && toNumType == t:NEVER 
-       && !t:isSubtypeSimple(t:intersect(reg.semType, t:NUMBER), toNumType) {
+    if toNumType != () && !t:isSubtypeSimple(t:intersect(reg.semType, t:NUMBER), toNumType) {
         toType = t:diff(toType, t:diff(t:NUMBER, toNumType));
-        bir:Register result = cx.createRegister(t:intersect(reg.semType, toType));
+        
+        bir:Register result = cx.createRegister(t:union(t:intersect(t:complement(t:NUMBER), reg.semType), toType));
```

Comment:
That doesn’t look right. Static type if `<T>E` is intersection of T and static type of E.

Reply:
If you are doing ConvertToInt on something with type T, then the type is int|(T - number)

Reply:
Compare with the type calculation in verify.bal.

Reply:
> int|(T - number)

This is what I was trying to do, I though `T - number` and `T ∩ ∁(number)` are the same. Let me give another try.

Reply:
>Compare with the type calculation in verify.bal.

Not sure what you meant by this, but I just added bir verifications for these two instructions.